crates/sparrow-syntax/src/syntax/fenl_type.rs

```diff
@@ -90,7 +92,7 @@ impl<'a> std::fmt::Display for FormatStruct<'a> {
 /// an instantiation of the signature. This leads to a relatively simple
 /// instantiation strategy, where a least upper bound of the types of the
 /// actual arguments is chosen for constrained type.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize)]
+#[derive(PartialOrd, Ord, Clone, Copy, Debug, PartialEq, Eq, Hash, Serialize)]
 #[repr(u8)]
 pub enum TypeConstraint {
```

Comment:
For follow up re: naming: I think this should be renamed to `TypeClass` to be more correct.